src/inmanta/deploy/scheduler.py

```diff
@@ -673,16 +675,28 @@ async def report_resource_state(
                     state.deployment_result = deployment_result
                 return
 
-            if skipped_for_dependency:
-                state.blocked = BlockedStatus.TRANSIENT
             # We are not stale
             state.status = status
             if deployment_result is not None:
                 # first update state, then send out events
                 self._deploying_latest.remove(resource)
                 state.deployment_result = deployment_result
                 self._work.finished_deploy(resource)
+                # Check if we need to mark a resource as transiently blocked
+                # Or if we can unblock a transiently blocked resource (its dependencies now succeed)
+                # We might already be unblocked if a dependency succeeded on another agent, e.g. while waiting for the lock
+                # so HandlerResourceState.skipped_for_dependency might be outdated
+                if state.blocked is not BlockedStatus.YES:
+                    if (
+                        resource_state is const.HandlerResourceState.skipped_for_dependency
+                        or state.blocked is BlockedStatus.TRANSIENT
+                    ):
+                        state.blocked = (
+                            BlockedStatus.NO if self._state.are_dependencies_compliant(resource) else BlockedStatus.TRANSIENT
+                        )
```

Comment:
I find this hard to understand.

What I think it means is 

```python
match state.blocked:
   case Yes: pass # blocked state can't be cleared here, I don't even think this code should be reachable?
   case Transient: 
        state.blocked = (
                            BlockedStatus.NO if self._state.are_dependencies_compliant(resource) else BlockedStatus.TRANSIENT
                        )
        # I don't understand how this is possible? the dependendencies would have updated us  no?
    case No:
           if resource_state is const.HandlerResourceState.skipped_for_dependency:
                      state.blocked = (
                            BlockedStatus.NO if self._state.are_dependencies_compliant(resource) else BlockedStatus.TRANSIENT
                        )
                     # I don't understand how self._state.are_dependencies_compliant(resource) can be false? Then we would have BlockedStatus.TRANSIENT?

```

Reply:
I updated this check to remove the `or state.blocked is BlockedStatus.Transient` following the discussion with @sanderr .
Regarding your questions about the inconsistency between `resource_state` and `are_dependencies_compliant` they were motivated by this:
> this deploy might run concurrently with a successful deploy of a dependency (if it's on another agent). If the dependency gets the scheduler lock first, we're already unblocked again by the time we reach this point.

More context: https://github.com/inmanta/inmanta-core/pull/8383#discussion_r1862230622